src/aggregations/top-hits-aggregation.js

```diff
@@ -0,0 +1,21 @@
+/**
+ * Internal counter for added top hits aggregations.
+ * @type {Number}
+ */
+let count = 0;
+
+/**
+ * Construct a Top hits aggregation.
+ *
+ * @param  {Object} opts  Options to include in the aggregation (maybe empty).
+ * @param  {String} name  Aggregation name. Defaults to agg_top_hits_{count}.
+ * @return {Object}       Top hits Aggregation.
+ */
+export default function topHitsAggregation(opts, name) {
+  name = name || `agg_top_hits_${count++}`
```

Comment:
> This is why in the tests the first aggregation is called agg_top_hits_0 and the second is agg_top_hits_1.
Exactly.

> even though they're not related to each other. 
Well... they kinda are. It's not really a matter of _who_ named the aggregation (whether it is the `top_hits` builder, the `AggregationBuilder` or some other module), but the fact that _this is the nth aggregation of the same type_ you are chaining.

So, what you're proposing is to remove the counter entirely and leave it up to users to rename the aggregation to avoid failure? I cannot say I'm sold on this vs. just maintaining a simple incremental variable. If this will be something to do "in the meantime", why don't just go along?